api/v1alpha2/istio_structs.go

```diff
@@ -86,6 +97,9 @@ type Resources struct {
 }
 
 type ResourceClaims struct {
-	Cpu    *string `json:"cpu,omitempty"`
+	// +kubebuilder:validation:Pattern=`^([0-9]+m?|[0-9]\.[0-9]{1,3})$`
+	Cpu *string `json:"cpu,omitempty"`
+
+	// +kubebuilder:validation:Pattern=`^[0-9]+(((\.[0-9]+)?(E|P|T|G|M|k|Ei|Pi|Ti|Gi|Mi|Ki|m)?)|(e[0-9]+))$`
```

Comment:
https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory
```suggestion
	// +kubebuilder:validation:Pattern=`^[0-9]+(((\.[0-9]+)?(E|P|T|G|M|k|Ei|Pi|Ti|Gi|Mi|Ki)?)|(e[0-9]+))$`
```